### PR TITLE
adds a dedicated SLD viewer, with callbacks on switches, feeders and vl navigation arrows

### PR DESCRIPTION
**Pull request by CBiasuzzi**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

adds a dedicated SLD viewer; other than rendering the SVG, it allows adding callbacks to the 
  switches and feeders display elements. Besides, when a third callback is defined, it adds arrows to the diagram that can be used, for example,  to navigate between voltage levels.
The demo illustrates its features (the example callbacks trigger messages in the console).

Note: this viewer originates in the gridstudy-app (original javascript code)

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by jonenst — DISMISSED** [review]

File: src/single-line-diagram-viewer.ts [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+
+export type OnFeederCallbackType = (
+    equipmentId: string,
+    equipmentType: string | null,
+    svgId: string,
+    x: number,
+    y: number
+) => void;
+
+export class SingleLineDiagramViewer {
```
Comment (etiennehomer) [inline src/single-line-diagram-viewer.ts:84]: Part of the constructor, the setters, the getters, init() and draw(), getDimensionsFromSvg() are duplicated from the network-area-diagram. 
Don't we want to try to mutualise them in a parent class ?

File: demo/src/index.js [hunk 2]
```diff
@@ -21,6 +25,89 @@ fetch(NadSvgExample)
...
+        document
+            .getElementById('svg-container-sld')
+            .getElementsByTagName('svg')[0]
+            .setAttribute('style', 'border:2px; border-style:solid;');
+    });
+
+const handleNextVL = (id) => {
+    const msg = 'Clicked on navigation arrow, dest VL is ' + id;
+    console.log(msg);
+    //alert(msg);
```
Comment (etiennehomer) [inline demo/src/index.js:60]: remove ?

File: demo/src/index.js [hunk 3]
```diff
@@ -21,6 +25,89 @@ fetch(NadSvgExample)
...
+const handleSwitch = (id, switch_status, element) => {
+    const msg =
+        'Clicked on switch: ' +
+        id +
+        ', switch_status: ' +
+        (switch_status ? 'close' : 'open') +
+        '. elementId: ' +
+        element.id;
+    console.log(msg);
+    //alert(msg);
```
Comment (etiennehomer) [inline demo/src/index.js:72]: remove ?

File: demo/src/index.js [hunk 4]
```diff
@@ -21,6 +25,89 @@ fetch(NadSvgExample)
...
+        ', feeder type: ' +
+        feederType +
+        ', svgId: ' +
+        svgId +
+        'x: ' +
+        x +
+        ', y: ' +
+        y;
+    console.log(msg);
+    //alert(msg);
```
Comment (etiennehomer) [inline demo/src/index.js:88]: remove ?

**Comment by etiennehomer** [comment]
I tested the integration with gridstudy, it works fine !

File: src/single-line-diagram-viewer.ts (hunk 1 above)
Comment (etiennehomer) [reply src/single-line-diagram-viewer.ts:84]: Discussed with Jon : we'll do it later. Code ok like this

**Review by etiennehomer — APPROVED** [review]

File: demo/src/index.js (hunk 2 above)
Comment (CBiasuzzi) [reply demo/src/index.js]: removed

File: demo/src/index.js (hunk 4 above)
Comment (CBiasuzzi) [reply demo/src/index.js]: removed

File: src/single-line-diagram-viewer.ts (hunk 1 above)
Comment (CBiasuzzi) [reply src/single-line-diagram-viewer.ts:84]: Thx, I'll remove the WIP tag, then

File: demo/src/index.js (hunk 3 above)
Comment (CBiasuzzi) [reply demo/src/index.js]: removed

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**